src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -57,6 +58,37 @@ internal static class DeserializationHelpers
             }
         }
 
+
+        internal static void ApplyInstanceAnnotations(object resource, IEdmStructuredTypeReference structuredType, ODataResourceWrapper resourceWrapper, 
```

Comment:
>ODataResourceWrapper [](start = 115, length = 20)

Since the only property of resourceWrapper you access is Resource (in line 89), you could just take in an ODataResource here (and call it with resourceWrapper.Resource.

Reply:
#Resolved